docs/4.2. Behaviour - Querying.md

```diff
@@ -2,63 +2,63 @@
 
-## Creating a Websocket subscription
+## Creating a WebSocket subscription
 
-In order to connect to a websocket, a client must first request a subscription of a particular type and with particular query parameters by performing a POST as defined in the [Query API](../APIs/QueryAPI.raml) subscriptions documentation. Use of GET requests to find existing suitable websocket connections is strongly discouraged.
+In order to connect to a WebSocket, a client MUST first request a subscription of a particular type and with particular query parameters by performing a POST as defined in the [Query API](../APIs/QueryAPI.raml) subscriptions documentation. Use of GET requests to find existing suitable WebSocket connections is strongly discouraged.
 
-There is no mandated URL base path for servers to use when creating websocket connections. Instead clients should observe the value of 'ws_href' which is returned by their subscription request in order to identify what to connect to.
+There is no mandated URL base path for servers to use when creating WebSocket connections. Instead clients SHOULD observe the value of 'ws_href' which is returned by their subscription request in order to identify what to connect to.
 
-Websockets created for specific clients will be cleaned up automatically if they disconnect.
+WebSockets created for specific clients will be cleaned up automatically if they disconnect.
 
-Connection upgrade procedures permitting an existing HTTP query to transition to a websocket are not currently supported, but may be in the future.
+Connection upgrade procedures permitting an existing HTTP query to transition to a WebSocket are not currently supported, but could be in the future.
 
 ### Query API Behaviour
-Long-lived connections to the Query API are supported via Websockets which can be set up via the /subscriptions resource. Query APIs may additionally support the HTTP 'Upgrade' header sent by clients to upgrade an HTTP GET request to a Websocket. In cases where this is performed, a corresponding entry in /subscriptions must also be created with matching query parameters.
 
-Upon receiving a request for a new subscription, the Query API should identify whether it already has any websockets open which provide for the requested query. If a websocket exists, the 'subscription' object representing it may be returned to the user. If a relevant websocket does not exist a new one should be created.
+Long-lived connections to the Query API are supported via WebSockets which can be set up via the /subscriptions resource. Query APIs MAY additionally support the HTTP 'Upgrade' header sent by clients to upgrade an HTTP GET request to a WebSocket. In cases where this is performed, a corresponding entry in /subscriptions MUST also be created with matching query parameters.
```

Comment:
Hmm, I thought we'd dealt with the fact that the sentence about HTTP 'Upgrade' here seems to cover precisely the thing that the previous section isn't currently supported... Need to check previous discussion.

Reply:
It was #149. Done.